### PR TITLE
Add index to commit_id in `GroupCommitResolution`

### PR DESCRIPTION
**Pull request by kjlundsgaard**
Goal is to avoid doing full sequence scanning on the `GroupCommitResolution` table in the `issues_resolved_via_release` endpoint

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/south_migrations/0342_auto__add_index_groupcommitresolution_commit_id.py [hunk 1]
```diff
@@ -0,0 +1,3245 @@
...
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import SchemaMigration
+from django.db import models
+
+
+class Migration(SchemaMigration):
+    def forwards(self, orm):
+        # Adding index on 'GroupCommitResolution', fields ['commit_id']
+        db.create_index('sentry_groupcommitresolution', ['commit_id'])
```
Comment (macqueen) [inline src/sentry/south_migrations/0342_auto__add_index_groupcommitresolution_commit_id.py]: so matt can correct me if this is unnecessary in this case, but generally we want to create indices concurrently for postgres. if you search for `CREATE INDEX CONCURRENTLY` there should be some examples where we've done that in the past for migrations

Comment (mattrobenolt) [inline src/sentry/south_migrations/0342_auto__add_index_groupcommitresolution_commit_id.py]: Yes please. All indexes should do this.

**Comment by mattrobenolt** [comment]
This migration is going to conflict with https://github.com/getsentry/sentry/pull/5836 for 0342, so whoever loses the race will need to change.

**Review by mattrobenolt — APPROVED** [review]

**Pull request merged**